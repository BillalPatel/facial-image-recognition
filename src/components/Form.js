import React from 'react';

const Form = (props) => {
  const {
    formName, displayNameField, nameChange, emailChange,
    buttonName, clickButton, passwordChange, confirmPassword,
    onPasswordChange, displaySignUpLink, submitSignUp
  } = props;

  return (
    <main className="pa4 black-80">
      <div className="measure center">
        <fieldset id="sign_up" className="ba b--transparent ph0 mh0">
          <legend className="f4 fw6 ph0 mh0">{formName}</legend>
          { displayNameField
          && (
          <div className="mt3">
            <label className="db fw6 lh-copy f6" htmlFor="name">Name</label>
            <input
              className="pa2 input-reset ba bg-transparent hover-bg-black hover-white w-100"
              type="email"
              name="firstname"
              id="firstname"
              onChange={nameChange}
            />
          </div>
          )}
          <div className="mt3">
            <label className="db fw6 lh-copy f6" htmlFor="email-address">Email Address</label>
            <input
              className="pa2 input-reset ba bg-transparent hover-bg-black hover-white w-100"
              type="email"
              name="email-address"
              id="email-address"
              onChange={emailChange}
            />
          </div>
          <div className="mv3">
            <div className="mt3">
              <label className="db fw6 lh-copy f6" htmlFor="password">Password</label>
              <input
                className="b pa2 input-reset ba bg-transparent hover-bg-black hover-white w-100"
                type="password"
                name="password"
                id="password"
                onChange={passwordChange}
              />
            </div>
            { confirmPassword
              && (
                <>
                  <div className="mt3" />
                  <label className="db fw6 lh-copy f6" htmlFor="password">Confirm Password</label>
                  <input
                    className="b pa2 input-reset ba bg-transparent hover-bg-black hover-white w-100"
                    type="password"
                    name="password"
                    id="confirm-password"
                    onChange={onPasswordChange}
                  />
                </>
              )}
          </div>
        </fieldset>
        <div>
          <input
            className="b ph3 pv2 input-reset ba b--black bg-transparent grow pointer f6 dib"
            id="submit-button"
            type="submit"
            value={buttonName}
            onClick={clickButton}
          />
        </div>
        { displaySignUpLink
         && (
         <div className="lh-copy mt3 hand">
           <p className="f6 link dim black db" id="signup-link" onClick={submitSignUp}>
            Sign up
           </p>
         </div>
         )}
      </div>
    </main>
  );
};

export default Form;
