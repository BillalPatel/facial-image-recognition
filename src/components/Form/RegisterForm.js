import React, { useState } from 'react';
import Form from './Form';

const RegisterForm = (props) => {
  const [name, setName] = useState('');
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');

  const {
    setUserName, onRouteChange
  } = props;


  const onNameChange = (event) => {
    setName(event.target.value);
  };

  const onEmailChange = (event) => {
    setEmail(event.target.value);
  };

  const onPasswordChange = (event) => {
    setPassword(event.target.value);
  };

  const onSubmitRegistration = async () => {
    await fetch('http://localhost:5000/register', {
      method: 'post',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify({
        name,
        email,
        password
      })
    })
      .then((res) => {
        if (res.status === 201) {
          setUserName(name);
          onRouteChange('landing');
        }
      });
  };

  const clickSignIn = () => {
    onRouteChange('signin');
  };

  return (
    <Form
      formName="Register"
      displayNameField
      nameChange={onNameChange}
      emailChange={onEmailChange}
      passwordChange={onPasswordChange}
      confirmPassword
      buttonName="Register!"
      clickButton={onSubmitRegistration}
      displaySignInLink
      displaySignUpLink={false}
      changeToSignUpForm={clickSignIn}
    />
  );
};

export default RegisterForm;
