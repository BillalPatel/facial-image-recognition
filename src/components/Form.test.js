/* eslint-disable no-unused-expressions */
import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow, mount, render } from 'enzyme';

import Form from './Form';

describe('Sign in form', () => {
  it('should render', () => {
    const formWrapper = mount(<Form />);
    expect(toJson(formWrapper)).toMatchSnapshot();
  });

  it('should render correctly as a login form', () => {
    const props = {
      displaySignUpLink: true,
      buttonName: 'Log in'
    };

    const formWrapper = shallow(<Form {...props} />);

    expect(formWrapper.find('#name').exists()).toEqual(false);
    expect(formWrapper.find('#email-address').exists()).toEqual(true);
    expect(formWrapper.find('#password').exists()).toEqual(true);
    expect(formWrapper.find('#confirm-password').exists()).toEqual(false);
    expect(formWrapper.find('#submit-button').prop('value')).toEqual('Log in');
    expect(formWrapper.find('#signup-link').exists()).toEqual(true);
  });

  it('should render correctly as a registration form', () => {
    const props = {
      displaySignUpLink: false,
      buttonName: 'Register!'
    };

    const formWrapper = shallow(<Form {...props} />);

    expect(formWrapper.find('#name').exists()).toEqual(true);
    expect(formWrapper.find('#email-address').exists()).toEqual(true);
    expect(formWrapper.find('#password').exists()).toEqual(true);
    expect(formWrapper.find('#confirm-password').exists()).toEqual(true);
    expect(formWrapper.find('#submit-button').prop('value')).toEqual('Register!');
    expect(formWrapper.find('#signup-link').exists()).toEqual(false);
  });
});
