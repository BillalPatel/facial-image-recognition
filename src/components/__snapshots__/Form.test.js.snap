// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign in form should render 1`] = `
<Form>
  <main
    className="pa4 black-80"
  >
    <div
      className="measure center"
    >
      <fieldset
        className="ba b--transparent ph0 mh0"
        id="sign_up"
      >
        <legend
          className="f4 fw6 ph0 mh0"
        />
        <div
          className="mt3"
        >
          <label
            className="db fw6 lh-copy f6"
            htmlFor="email-address"
          >
            Email Address
          </label>
          <input
            className="pa2 input-reset ba bg-transparent hover-bg-black hover-white w-100"
            id="email-address"
            name="email-address"
            type="email"
          />
        </div>
        <div
          className="mv3"
        >
          <div
            className="mt3"
          >
            <label
              className="db fw6 lh-copy f6"
              htmlFor="password"
            >
              Password
            </label>
            <input
              className="b pa2 input-reset ba bg-transparent hover-bg-black hover-white w-100"
              id="password"
              name="password"
              type="password"
            />
          </div>
        </div>
      </fieldset>
      <div>
        <input
          className="b ph3 pv2 input-reset ba b--black bg-transparent grow pointer f6 dib"
          id="submit-button"
          type="submit"
        />
      </div>
    </div>
  </main>
</Form>
`;
